addons/l10n_it_edi/wizard/account_move_send.py

```diff
@@ -89,9 +104,12 @@ def _call_web_service_after_invoice_pdf_render(self, invoices_data):
             attachments_vals = {}
             moves = self.env['account.move']
             for move in invoices_data:
-                if self._get_default_l10n_it_edi_enable_send(move):
+                if move._l10n_it_edi_ready_for_xml_export():
                     moves |= move
-                    attachments_vals[move] = invoices_data[move]['l10n_it_edi_values']
+                    if attachment := move.l10n_it_edi_attachment_id:
```

Comment:
how could that happen?

Reply:
This happens when you generated the XML, and you re-select the invoices and want to send them with the dialog.
In that case, the XML is already there, so  it's not in `invoices_data`, but it's on the move already.

It's literally the entire purpose of the PR. 😅  We can generate and send in two different moments.

Reply:
The method name is still confusing then, because also the PDF should have been generated already.

Reply:
It's called like that in `account`... I just extend it.